### ISSUE TITLE
Can't use ng-model with empty object since 1.3.x update

### ISSUE THREAD

**Issue by chatellier** [zensh/ui-autocomplete#18]
Since 1.3.x update, ui-autocomplete component can't use empty ng-model object.

This can be seen here : http://jsfiddle.net/ar1bqtat/3/

Error is : 
```TypeError: Cannot read property 'value' of undefined
    at Object.fn (autocomplete-0.5.7.js:137:44)
    at Scope.$digest (https://code.angularjs.org/1.3.4/angular.js:14129:29)
    at Scope.$apply (https://code.angularjs.org/1.3.4/angular.js:14391:24)
    at bootstrapApply (https://code.angularjs.org/1.3.4/angular.js:1437:15)
    at Object.invoke (https://code.angularjs.org/1.3.4/angular.js:4138:17)
    at doBootstrap (https://code.angularjs.org/1.3.4/angular.js:1435:14)
    at bootstrap (https://code.angularjs.org/1.3.4/angular.js:1455:12)
    at angularInit (https://code.angularjs.org/1.3.4/angular.js:1349:5)
    at HTMLDocument.<anonymous> (https://code.angularjs.org/1.3.4/angular.js:25910:5)
    at j (http://code.jquery.com/jquery-1.11.1.min.js:2:27244)```

**Comment by chamarts** [comment]
Hi, is there any update on this issue ? I am having the same issue.

**Comment by chamarts** [comment]
TypeError: Cannot read property 'value' of undefined
    at Object.fn (autocomplete.js:138)
    at Scope.$get.Scope.$digest (angular.js:14225)
    at Scope.$get.Scope.$apply (angular.js:14488)
    at done (angular.js:9646)
    at completeRequest (angular.js:9836)
    at XMLHttpRequest.requestLoaded (angular.js:9777)

**Comment by zensh** [comment]
I have test with v1.3.5, it work well

**Comment by chamarts** [comment]
Hi, do you have jFiddle/plunkr example ?

**Comment by zensh** [comment]
`There must have a property **'value'** in object type ngModel.`

https://github.com/zensh/ui-autocomplete/tree/master/demo

**Comment by chamarts** [comment]
There is one. but when I set the value="{{}}" in the input element, it is
showing as Object [][] when it loads

The initial item is $scope.item = {
                              txtName : {
                                    id : '',  value : '', name : ''
                              }
                        }

**Comment by chamarts** [comment]
I did some debugging and  looks like the issue is with this code at autocomplete: Line 137

              scope.$watch(attr.ngModel, function (model) {
                if (valueMethod() !== model.value) {
                  ctrl.$viewValue = model.value;
                  ctrl.$render();
                }
              }, true);

model is always undefined in this function. I think the $watch method takes a different function with function (newVal, oldValue, x) 

the model here should actually be ngModel object. I fixed it in my local and it works now with initializing data on load.
